File: src/main/php/memcache_client.php
<?php

echo "starting client test\n";

$host = "localhost";
$port = 11211;
$key = "HelloWorld";
$expiry = 300;

$shortopts = "";
$longopts  = array(
    "host::",
    "port::",
    "key::",
    "value::",
    "expiry::"
);

$options = getopt($shortopts, $longopts);

if (isset($options['host'])) {
    echo "setting host to {$options['host']}\n";
    $host = $options['host'];
}

if (isset($options['port'])) {
    echo "setting port to {$options['port']}\n";
    $port = $options['port'];
}

if (isset($options['key'])) {
    echo "setting key to {$options['key']}\n";
    $key = $options['key'];
}

if (isset($options['expiry'])) {
    echo "setting expiry to {$options['expiry']}\n";
    $expiry = $options['expiry'];
}

echo "connecting to memcache host={$host} port={$port}\n";
$memcache_client = new Memcached();
$memcache_client->addServer($host, $port);

if (isset($options['value'])) {
    $status = $memcache_client->set($key, $options['value'], intval($expiry));
    echo "setting key=[{$key}] to value=[{$options['value']}] expire=[{$expiry}] status=[{$status}]\n";

}

$v = $memcache_client->get($key);
echo "key=[{$key}] value=[{$v}]\n";

File: src/main/python/memcache_client.py
import argparse

from pymemcache.client import base

parser = argparse.ArgumentParser()

parser.add_argument("--host",
                    action='store',
                    help="echo the string you use here",
                    type=str, default="localhost")

parser.add_argument("--port",
                    action='store',
                    help="echo the string you use here",
                    type=int, default=11211)

parser.add_argument("--key",
                    action='store',
                    help="echo the string you use here",
                    type=str, default="MyTest")

parser.add_argument("--value",
                    action='store',
                    help="echo the string you use here",
                    type=str)

parser.add_argument("--expiry",
                    action='store',
                    help="echo the string you use here",
                    type=int, default=300)

args = parser.parse_args()

print("Connecting to memcache host={} port={}".format(args.host, args.port))
client = base.Client((args.host, args.port))

if args.value:
    print("setting key={} value={}".format(args.key, args.value))
    client.set(args.key, args.value, expire=args.expiry)

print("getting value for key={}".format(args.key))
value = client.get(args.key)

print("key=[{}] value=[{}]".format(args.key, args.value))
